ecr/ref.go

```diff
@@ -35,6 +35,10 @@ const (
 var (
 	invalidARN = errors.New("ref: invalid ARN")
 	splitRe    = regexp.MustCompile(`[:@]`)
+	// Expecting to match ECR image names of the form:
+	// Example 1: 777777777777.dkr.ecr.us-west-2.amazonaws.com/my_image:latest
+	// Example 2: 777777777777.dkr.ecr.cn-north-1.amazonaws.com.cn/my_image:latest
+	ecrRegex = regexp.MustCompile(`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?.*`)
```

Comment:
This doesn't currently support the [ECR FIPS endpoints](https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-endpoints.html), which will have "ecr-fips" in the URL instead of "ecr".

It's fine if you don't want to support that right now, but I'd appreciate at least a TODO comment.